src/main.c

```diff
@@ -73,6 +74,62 @@ static int set_all_thread_affinity(void)
     return 0;
 }
 
+static void dpvs_usage(const char *prgname)
+{
+    printf("\nUsage: %s ", prgname);
+    printf("DPVS application options:\n"
+            "   -v  version     display DPVS version info\n"
+            "   -h  help        display DPVS help info\n"
+    );
+}
+
+static int parse_app_args(int argc, char **argv)
+{
+    const char *short_options = "vh";
+    char *prgname = argv[0];
+    int c, ret = -1;
+
+    const int old_optind = optind;
+    const int old_optopt = optopt;
+    char * const old_optarg = optarg;
```

Comment:
`const char *old_optarg = ...`

Reply:
char *const is a const pointer to char while const char * is a pointer to const char. We need the pointer not the value to be unchanged. We can also see it at eal_parse_args.